stdlib/public/core/StringUTF8.swift

```diff
@@ -48,8 +48,10 @@ extension _StringCore {
         src: UnsafeMutablePointer(startASCII + i),
         size: numericCast(utf16Count))
 
-      return (i + utf16Count, result)
+      // Convert the _UTF8Chunk into host endianness.
+      return (i + utf16Count, _UTF8Chunk(littleEndian: result))
```

Comment:
@rintaro You are right about `_nthContiguous()` being little-endian only but that is only part of a larger problem. There does not seem to be any support for UTF-16BE in Swift, and supporting it may require considerable changes to the _StringCore code. I am pretty sure that conversion from UTF-8 to UTF-16 is currently broken in this port, but we hope to tackle that in a future pull request (soonish).